src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ModelBindingHelper.cs

```diff
@@ -636,17 +636,16 @@ internal static string GetPropertyName(Expression expression)
         /// Clears <see cref="ModelStateDictionary"/> entries for <see cref="ModelMetadata"/>.
         /// </summary>
         /// <param name="modelMetadata">The <see cref="ModelMetadata"/>.</param>
+        /// <param name="modelstate">The <see cref="ModelStateDictionary"/>.</param>
         /// <param name="modelKey">The entry to clear. </param>
-        /// <param name="modelMetadataProvider">The <see cref="IModelMetadataProvider"/>.</param>
         public static void ClearValidationStateForModel(
```

Comment:
There are really only three cases w/ an empty prefix:
- A simple type: Just remove the `""` entry, if any.
- A collection: Remove anything w/ a key that starts w/ `'['`.
- A POCO: Remove anything that's a prefix match for a property in the type.

That's fairly precise.